README.md

```diff
@@ -1,27 +1,305 @@
+## Installing the SDK
 
-Run the following code in your project. These commands will add the Descope ExpresSDK for Python as a project dependency, and set the `DESCOPE_PROJECT_ID` variable to a valid \<ProjectID\>.
+Install the package with:
 
-```code
+```bash
 pip install descope
-export DESCOPE_PROJECT_ID=<ProjectID>
 ```
 
-## What do you want to implement?
+## Setup
+
+A Descope `Project ID` is required to initialize the SDK. Find it on the
+[project page in the Descope Console](https://app.descope.com/settings/project).
+
+```python
+from descope import DescopeClient
+
+# Initialized after setting the DESCOPE_PROJECT_ID env var
+descope_client = DescopeClient()
+
+# ** Or directely **
+descope_client = DescopeClient(project_id="<Project ID>")
+```
+
+## Usage
+
+Here are some examples how to manage and authenticate users:
+
+### OTP Authentication
+
+Send a user a one-time password (OTP) using your preferred delivery method (_email, SMS, Whatsapp message_). An email address or phone number must be provided accordingly.
+
+The user can either `sign up`, `sign in` or `sign up or in`
+
+```python
+from descope import DeliveryMethod
+
+# Every user must have an identifier. All other user information is optional
+email = "desmond@descope.com"
+user = {"name": "Desmond Copeland", "phone": "212-555-1234", "email": email}
+descope_client.otp.sign_up(method=DeliveryMethod.EMAIL, identifier=email, user=user)
+```
+
+The user will receive a code using the selected delivery method. Verify that code using:
+
+```python
+jwt_response = descope_client.otp.verify_code(
+    method=DeliveryMethod.EMAIL, identifier=email, code=value
+)
+```
+
+The session and refresh JWTs should be returned to the caller, and passed with every request in the session. Read more on [session validation](#session-validation)
+
+### Magic Link
+
+Send a user a Magic Link using your preferred delivery method (_email, SMS, Whatsapp message_).
+The Magic Link will redirect the user to page where the its token needs to be verified.
+This redirection can be configured in code, or generally in the [Descope Console](https://app.descope.com/settings/authentication/magiclink)
+
+The user can either `sign up`, `sign in` or `sign up or in`
+
+```python
+from descope import DeliveryMethod
+
+descope_client.magiclink.sign_up_or_in(
+    method=DeliveryMethod.EMAIL,
+    identifier="desmond@descope.com",
+    uri="http://myapp.com/verify-magic-link", # Set redirect URI here or via console
+)
+```
+
+To verify a magic link, your redirect page must call the validation function on the token (`t`) parameter (`https://your-redirect-address.com/verify?t=<token>`):
+
+```python
+jwt_response = descope_client.magiclink.verify(token=token)
+```
+
+The session and refresh JWTs should be returned to the caller, and passed with every request in the session. Read more on [session validation](#session-validation)
```

Comment:
it feels a bit redundant to mention this on every auth method IMO, i think saying it once is enough

but worth to ask for other feedback on this

Reply:
I'm leaving it in, the reasoning being, that probably most of readers won't go through all of the README, but search for specific auth methods. I want their journey to be complete. But sure, if people find it redundant we can remove it.